service/pki/service.go

```diff
@@ -230,25 +163,6 @@ func (s *service) Create(config CreateConfig) error {
 		}
 	}
 
-	// Create a role for the mounted PKI backend, if it does not already exist.
-	created, err := s.IsRoleCreated(config.ClusterID)
-	if err != nil {
-		return microerror.Mask(err)
-	}
-	if !created {
-		data := map[string]interface{}{
-			"allowed_domains":    config.AllowedDomains,
-			"allow_subdomains":   "true",
-			"ttl":                config.TTL,
-			"allow_bare_domains": config.AllowBareDomains,
-		}
-
-		_, err = logicalBackend.Write(s.WriteRolePath(config.ClusterID), data)
-		if err != nil {
-			return microerror.Mask(err)
-		}
-	}
```

Comment:
I miss how this works now. I was under the impression the PKI backend has to be attached to a role. The fact that we create roles on the fly for each and every key pair is fine, but I am not sure this implies we should not create a dedicated role for the PKI. AFAIK several attributes are defined by the role the PKI is associated with. Not doing this might cause the PKI to be kind of orphaned. Uncertain here but feels like too many changes at once and I am quite uncomfortable with this at this point because I am not into all the details.